src/git/github-git/repo-tags.ts

```diff
@@ -0,0 +1,102 @@
+import { execWithOutput } from "../../utils";
+
+export interface GitTag {
+  name: string;
+  ref: string;
+}
+
+/**
+ * Pushes any missing tags to the remote first replacing those tags with lightweight versions.
+ * We replace annotated tags with lightweight ones because we cannot sign annotated tags, but
+ * lightweight tags pointing to signed commits will show up as verified in GitHub.
+ */
+export async function pushTags(cwd?: string) {
+  const onlyLocalTags = await getOnlyLocalTags(cwd);
+
+  await deleteTags(onlyLocalTags, cwd);
+  const createdTags = await createLightweightTags(onlyLocalTags, cwd);
+  await execWithOutput("git", ["push", "origin", "--tags"], { cwd });
+
+  return createdTags;
+}
+
+export async function getOnlyLocalTags(cwd?: string): Promise<GitTag[]> {
+  const localTags = await getLocalTags(cwd);
+  // Checkout action uses origin for the remote
+  // https://github.com/actions/checkout/blob/main/src/git-source-provider.ts#L111
+  const remoteTagNames = await getRemoteTagNames("origin", cwd);
+
+  const diff = computeTagDiff(localTags, remoteTagNames);
+
+  return diff;
+}
+
+export async function createLightweightTags(
+  tags: GitTag[],
+  cwd?: string
+): Promise<GitTag[]> {
+  const createdTags = tags.map(async (tag) => {
+    await execWithOutput("git", ["tag", tag.name, tag.ref], { cwd });
+
+    return tag;
+  });
+
+  return await Promise.all(createdTags);
+}
+
+export async function deleteTags(tags: GitTag[], cwd?: string) {
+  const deleteCommands = tags.map(async (tag) => {
+    await execWithOutput("git", ["tag", "-d", tag.name], { cwd });
+    return tag;
+  });
+  await Promise.all(deleteCommands);
+
+  return deleteCommands;
```

Comment:
Is below equivalent? 

```suggestion
  return await Promise.all(deleteCommands);
```